### PR TITLE
Action for adding global strain objects

### PR DESCRIPTION
**Pull request by SudiptaBiswas**
Action for adding all the objects associated with global strain and corresponding displacement calculation. 

Refs #11314

### CODE REVIEW

File: modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GlobalStrain/GlobalStrainAction.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+- [Global Strain ScalarKernel](/GlobalStrain.md): Solves for the global strain in terms of the scalar variable
+
+### Material
+
+- [Global Strain Material](/ComputeGlobalStrain.md): Calculates the global strain components from the scalar variable
+
+### AuxVariables
+
+- [Global Displacement AuxKernels](/GlobalDisplacementAux.md): Creates AuxVariables for each additional displacement components
```
Comment (dschwen) [inline modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GlobalStrain/GlobalStrainAction.md]: Do you really want to link to the kernel here, too?

File: modules/tensor_mechanics/src/actions/GlobalStrainAction.C [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  params.addRequiredParam<NonlinearVariableName>("scalar_global_strain",
+                                                 "Scalar variable for global strain");
+  params.addParam<std::vector<VariableName>>("displacements", "The displacement variables");
+  params.addParam<std::vector<AuxVariableName>>(
+      "auxiliary_displacements",
+      "The auxliary displacement variables to be calculated from scalar variables");
+  params.addParam<std::vector<Real>>("applied_stress_tensor",
+                                     "Vector of values defining the constant applied stress "
+                                     "to add, in order 11, 22, 33, 23, 13, 12");
+  params.addParam<bool>("use_displaced_mesh", false, "Whether to use displaced mesh");
```
[the remark this answers is not in the document]
Comment (SudiptaBiswas) [reply modules/tensor_mechanics/src/actions/GlobalStrainAction.C]: No, I guess I can remove it from input parameters and set then individually for the objects.

File: modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GlobalStrain/index.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  [./block_b]
+    ...
+  [../]
+[]
+```
+
+## Toplevel parameters
+
+Parameters supplied at the `[Modules/TensorMechanics/GlobalStrain]` level act as
+defaults for the Master action subblocks.
```
[the remark this answers is not in the document]
Comment (permcody) [reply modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GlobalStrain/index.md]: Right, we will likely add this in the future. It's not implemented today.

File: modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GlobalStrain/GlobalStrainAction.md (hunk 1 above)
Comment (SudiptaBiswas) [reply modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GlobalStrain/GlobalStrainAction.md]: The documentation for how to create aux variables for visualization is within `GlobalDisplacementAux.md`. So, I wanted to provide same link. The text should say `AuxVariables` though.

File: modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GlobalStrain/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (SudiptaBiswas) [reply modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GlobalStrain/index.md]: Okay, will change the documentation.

**Review by dschwen — APPROVED** [review]

**Pull request merged**